behavior/MouseTracking/get_mouse_XY_pos.m

```diff
@@ -0,0 +1,144 @@
+        frame_range = frame_chunks(idx,:);
+        video = read(behavior_vid,frame_range);
+%         video = squeeze(video(:,:,1,:)); % 3D movie stack
+        
+        if display_tracking
+            
+            % plot original image on the left
+            subplot(121);
+            image = video(:,:,:,1);
+            h = imagesc(image);
+            title(sprintf('Original: Frames %d - %d',...
+                          frame_chunks(idx,1), frame_chunks(idx,2)));
+            axis image; colormap gray;
+            hold on
+            
+            % plot initial centroids on original video
+            k = plot(0,0,'k*');
+
+            % plot initial tracking image on the right
+            subplot(122);
+            j = imagesc(image); colormap gray;
+            title(sprintf('Tracker: Frames %d - %d',...
+                          frame_chunks(idx,1),frame_chunks(idx,2)));
+            hold on
+            axis image;
+
+            % plot centroids on tracking video
+            l = plot(0,0,'k*');
+        end
+
+        for frame_idx = 1:size(video,4)
+           
+           % Update original image CData
+            image = video(:,:,:,frame_idx);
+%             image = im2double(image,'indexed');
+            if display_tracking
+                set(h,'CData',image);
+                pause(0.00001);
+            end
+           
+            % Find the mouse blob using findMouse helper function
+            thresh = 20; % assumes that black mouse has RGB values <20
+            [final_image,s] = findMouse(bg_image,image,thresh);
+            area_vector = [s.Area];
+            length_vector = [s.MinorAxisLength];
+            
+            if display_tracking
+                % Update tracking
+                set(j,'CData',final_image);
+                pause(0.00001);
+            end
+
+            % Save centroids data and update plot
+            if isempty(area_vector) %sometimes blob disappears, go to previous centroid
+                centroids(frame_chunks(idx,1)+frame_idx-1,:) = c_old;
+                
+                if display_tracking
+                    % Update subplots
+                    set(k,'XData',c_old(1),'YData',c_old(2),'color','r');
+                    set(l,'XData',c_old(1),'YData',c_old(2),'color','r');
+                    pause(0.00001);
+                end
+                
+            else % new centroid
+                [~, id] = max(length_vector); %assume mouse is the fattest blob
+                c_new = s(id(1)).Centroid(1:2);
+                centroids(frame_chunks(idx,1)+frame_idx-1,:) = c_new;
+                c_old = c_new;
+                
+                if display_tracking
+                    % Update subplots
+                    set(k,'XData',c_new(1),'YData',c_new(2),'color','b');
+                    set(l,'XData',c_new(1),'YData',c_new(2),'color','b');
+                    pause(0.00001);
+                end
```

Comment:
If you pull out this code chunk `if display_tracking ... end` out of the containing `if isempty(area_vector) ... end`, then you don't have to repeat the display block.